integration-tests/biometric-provider-truvera.test.ts

```diff
@@ -0,0 +1,296 @@
+      
+      const storedMatchCredential = matchCredentials[0];
+      expect(storedMatchCredential.type).toContain(biometricConfigs.biometricMatchCredentialType);
+      expect(storedMatchCredential.credentialSubject.biometric.data).toBe('hidden');
+    }, 30000);
+  });
+
+  describe('Matching process', () => {
+    it('should create only match credential when enrollment credential exists', async () => {
+      // Ensure there's an enrollment credential
+      const credentialProvider = getCredentialProvider();
+      let enrollmentCredentials = await credentialProvider.getCredentials(biometricConfigs.enrollmentCredentialType);
+      
+      // If there's no enrollment credential, create one first
+      if (enrollmentCredentials.length === 0) {
+        const walletDID = await getDIDProvider().getDefaultDID();
+        const proofRequest = {
+          input_descriptors: [
+            {
+              constraints: {
+                fields: [
+                  {
+                    path: ['$.credentialSubject.biometric.id'],
+                  },
+                  {
+                    path: ['$.credentialSubject.biometric.created'],
+                  },
+                ],
+              },
+            },
+          ],
+        };
+        
+        const result = await idvProvider.enroll(walletDID, proofRequest);
+        await credentialProvider.addCredential(result.enrollmentCredential);
+        enrollmentCredentials = await credentialProvider.getCredentials(biometricConfigs.enrollmentCredentialType);
+      }
+      
+      // Remove any existing match credentials
+      const existingMatchCredentials = await credentialProvider.getCredentials(biometricConfigs.biometricMatchCredentialType);
+      for (const credential of existingMatchCredentials) {
+        await credentialProvider.removeCredential(credential.id);
+      }
+      
+      const proofRequest = {
+        input_descriptors: [
+          {
+            constraints: {
+              fields: [
+                {
+                  path: ['$.credentialSubject.biometric.id'],
+                },
+                {
+                  path: ['$.credentialSubject.biometric.created'],
+                },
+              ],
+            },
+          },
+        ],
+      };
+      
+      // Start the IDV process
+      const startIDVPromise = biometricProvider.startIDV(proofRequest);
+      
+      // Wait for async operations to complete
+      await new Promise(resolve => setTimeout(resolve, 100));
+      
+      // Simulate the completion event ourselves (in real app, the wallet would do this)
+      const walletDID = await getDIDProvider().getDefaultDID();
+      const result = await idvProvider.match(walletDID, enrollmentCredentials[0], proofRequest);
+      biometricProvider.eventEmitter.emit('onComplete', result);
+      
+      // Wait for startIDV to finish
+      await startIDVPromise;
```

Comment:
feels like the lines above this are repeated - any chance to pull out a helper method?